### ISSUE TITLE
NetworkPolicy bug for range 10.0.0.0/8

### ISSUE THREAD

**Issue by aminmr** [kubeovn/kube-ovn#3193]
# Expected Behavior
After applying the network policy with my policies, everything works fine, and blocking or unblocking any IP in range 10.0.0.0/8 works properly.

# Actual Behavior
When I block all 10.0.0.0/8 except 10.96.0.10(CoreDNS) and 10.96.0.1(ApiServer), the blocking works fine, but it will not make the exception. This policy is still working on version 1.10.10, but after updating the kube-OVN to 1.12.0, this problem occurred.

# Steps to Reproduce the Problem

1. Install kube-ovn with script or helm version 1.10.10 and apply the following network policy:
```
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
spec:
  egress:
  - to:
    - ipBlock:
        cidr: 10.96.0.10/32
  - to:
    - ipBlock:
        cidr: 10.96.0.1/32
  - to:
    - ipBlock:
        cidr: 0.0.0.0/0
        except:
        - 10.0.0.0/8
        - 192.168.1.0/24
  podSelector: {}
  policyTypes:
  - Egress
```
2. It should block access to all IPs in range 10.0.0.0/8 except the 10.96.0.10/32 and 10.96.0.1/32
3. Now upgrade the kube-OVN version to 1.12.0 and check if this network policy works. It will not!

# Additional Info
1. For solving coreDNS problem temporary I used the following way to access coreDNS pods:
```
  - to:
    - namespaceSelector:
        matchLabels:
          kubernetes.io/metadata.name: kube-system
      podSelector:
        matchLabels:
          k8s-app: kube-dns
```
2. I checked this rule for range 192.168.1.0/24 and made an exception for specific IP and this problem didn't occured.

- Kubernetes version:
`1.25.6`

  **Output of `kubectl version`:**

```  
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.4", GitCommit:"fa3d7990104d7c1f16943a67f11b154b71f6a132", GitTreeState:"archive", BuildDate:"2023-07-20T07:37:53Z", GoVersion:"go1.20.6", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.6", GitCommit:"ff2c119726cc1f8926fb0585c74b25921e866a28", GitTreeState:"clean", BuildDate:"2023-01-18T19:15:26Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"linux/amd64"}
```
- **kube-ovn version**:
`1.12.0`

- operation-system/kernel version:
  
  **Output of `awk -F '=' '/PRETTY_NAME/ { print $2 }' /etc/os-release`:**
```
"Debian GNU/Linux 11 (bullseye)"
```
  **Output of `uname -r`:**
```
5.10.0-25-amd64
```

**Comment by aminmr** [comment]
For further information, I tested this on version 1.11.10, and this problem occurred there. If you have any plans to patch 1.11.x You can patch this PR on this version, too @zhangzujian
Thanks in advance.

**Comment by zhangzujian** [comment]
This issue was introduced in #2251 for Kube-OVN >= v1.12.0 and #2355 for Kube-OVN >= v1.11.2.

Once fixed, the patch will be merged into v1.11 and v1.12.

**Comment by changluyi** [comment]
the networkpolicy rule is wrong, you should allow the service'backend ip  but not service ip
![image](https://github.com/kubeovn/kube-ovn/assets/47097611/2ecf37f9-57c2-4b22-9ed9-2418e1d5ef1c)


you can also try on calico,  calico and kube-ovn have the same behaviour

**Comment by aminmr** [comment]
Dear @changluyi 
I checked this networkPolicy with Calico version v3.22.0 and works correctly. Which version of Calico did you test this networkPolicy?
